File: python/Arrays/linear_search.py
def linear_search(arr, target):
    """
    Linear search algorithm
    Time Complexity: O(n)
    Space Complexity: O(1)
    """
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

def main():
    # Read input
    n = int(input())
    arr = list(map(int, input().split()))
    target = int(input())
    
    # Perform search
    result = linear_search(arr, target)
    
    # Print result
    print(result)

if __name__ == "__main__":
    main() 

File: cpp/Arrays/LinearSearch.cpp
#include <iostream>
#include <vector>
using namespace std;

int linearSearch(vector<int> &vec, int target)
{
    for(int i=0; i<vec.size(); i++) {
        if(vec[i] == target) {
            return i;
        }
    }
    return -1;
}

int main()
{
    int n;
    cin >> n;

    vector<int> vec(n);
    for (int i = 0; i < n; i++)
    {
        cin >> vec[i];
    }

    int target;
    cin >> target;

    int result = linearSearch(vec, target);

    cout << result << endl;
    return 0;
}
